src/main/java/org/elasticsearch/index/mapper/internal/ParentFieldMapper.java

```diff
@@ -316,6 +365,10 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     @Override
     public void merge(Mapper mergeWith, MergeResult mergeResult) throws MergeMappingException {
         ParentFieldMapper other = (ParentFieldMapper) mergeWith;
+        if (parent != other.parent) {
```

Comment:
since parent is an object, can you use .equals instead? (or Objects.equal like for type)